### PR TITLE
update request.js to match new spec changes

### PR DESCRIPTION
**Pull request by KhafraDev**
Implements spec changes from https://github.com/whatwg/fetch/commit/8e4553bcc338fd6db98cc6ba60fc03cea48fad4c and also fixes a few spec divergences (I'll comment on each one to make it easier)

refs: #1141

### CODE REVIEW

File: lib/fetch/request.js [hunk 1]
```diff
@@ -138,8 +138,47 @@ class Request {
 
     // 12. Set request to a new request with the following properties:
     request = makeRequest({
```
Comment (KhafraDev) [inline lib/fetch/request.js:140]: by listing all the properties, I assumed there are some that aren't meant to be copied to the new request. It makes sense as previously origin and urlList were not supposed to be copied, but were being done by the lib.

File: lib/fetch/request.js [hunk 2]
```diff
@@ -138,8 +138,47 @@ class Request {
...
-      ...request,
-      window
+      // URL request’s URL.
```
Comment (KhafraDev) [inline lib/fetch/request.js:141]: this was previously defined as the [current url](https://fetch.spec.whatwg.org/#concept-request-current-url) which is the last url in the urlList, but the recent spec changes technically fixed this on behalf of undici :)

**Review by ronag — APPROVED** [review]

**Comment by ronag** [comment]
Good work!

File: lib/fetch/request.js [hunk 3]
```diff
@@ -138,8 +138,47 @@ class Request {
...
+      // undici implementation note: this is set as the first item in request's urlList in makeRequest
+      // method request’s method. 
+      method: request.method,
+      // header list A copy of request’s header list.
+      headersList: request.headersList,
```
Comment (ronag) [inline lib/fetch/request.js:146]: Is this also cloned in makeRequest?

Comment (KhafraDev) [reply lib/fetch/request.js:146]: it is https://github.com/nodejs/undici/blob/main/lib/fetch/request.js#L753-L755

Comment (ronag) [reply lib/fetch/request.js:146]: Maybe add a comment like you did below for urlList?

**Comment by ronag** [comment]
@KhafraDev excellent work, if you see any other spec discrepancies feel free to open PR!

**Pull request merged**